packages/clay-badge/src/ClayBadge.tsx

```diff
@@ -0,0 +1,34 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+import * as React from 'react';
+import classNames from 'classnames';
+
+interface props {
```

Comment:
I assume a good chunk of this would be common... I know this is likely a premature abastraction, but isn't there something like a declared base interface for this so in cases you don't want to declare anything else you can avoid the additional interface, and in case you want to add something, you just need to specify those?

Reply:
I'm not totally sure I am understanding your question here. You want to abstract the interface here?

Reply:
Maybe this is classic early-abstraction, so it can wait. I was thinking...

```
// @types/baseprops.dts
export const interface baseprops {
    // not sure if there are other cases than the prop index
    [propName: string]: any;
}
```

```
import baseprops from '@types/baseprops';

interface props extends baseprops {
    className?: string;
    label?: string | number;
    type?: string;
}

const ClayBadge: React.FunctionComponent<props> = ({
...
});
```

Reply:
I might've jumped the gun because of `className`, so it's likely I'm missing something, but just in case :)

Reply:
`[propName: string]: any;` would essentially be a catch all for when people want to allow "...otherProps". Let me look into how the community handles this and maybe there is a more simple way. It may even be an anti-pattern, and instead we should extend the interface from the html element the `otherProps` are being applied to.

Reply:
Rather than using the `any` type, we should extend from the HTMLElement's attributes interface.

See my update https://github.com/liferay/clay/pull/1548/commits/4e28a0e64e1b606b81bae796f7a462c2c2c2610d